vignettes/datatable-fread-and-fwrite.Rmd

```diff
@@ -0,0 +1,325 @@
+
+After applying any `skip` or `nrows` settings (if specified), the first row with a consistent number of fields is examined:
+
+If all fields in this line are interpretable as character and the values do not strongly resemble a data row (e.g., a row of purely numeric-looking strings might still be considered data), it is typically used as the header (column names).
+
+Otherwise (e.g., if the line contains detected numeric types, or character strings that strongly resemble numbers and could be data), it is treated as a data row, and default column names (`V1`, `V2`, …) are assigned.
+
+You can explicitly tell fread whether a header exists using `header = TRUE` or `header = FALSE`.
+
+**Skip Detection**
+
+By default (`skip="auto"`), `fread` will automatically skip blank lines and comment lines (e.g., starting with `#`) before the data header. To manually skip a specific number of lines, use `skip=n`.
+
+### 1.3 High-Quality Automatic Column Type Detection
+
+Many real-world datasets contain columns that are initially blank, zero-filled, or appear numeric but later contain characters. To handle such inconsistencies, `fread()` in `data.table` employs a robust column type detection strategy.
+
+Since v1.10.5, `fread()` samples rows by reading blocks of contiguous rows from multiple equally spaced points across the file, including the start, middle, and end. The total number of rows sampled is chosen dynamically based on the file size and structure, and is typically around 10,000, but can be smaller or slightly larger. This wide sampling helps detect type changes that occur later in the data (e.g., `001` to `0A0` or blanks becoming populated).
+
+**Efficient File Access with mmap**
+
+To implement this sampling efficiently, `fread()` uses the operating system's memory-mapped file access (`mmap`), allowing it to jump to arbitrary positions in the file without sequential scanning. This lazy, on-demand strategy makes sampling nearly instantaneous, even for very large files.
+
+If a jump lands within a quoted field that includes newlines, `fread()` tests subsequent lines until it finds 5 consecutive rows with the expected number of fields, ensuring correct parsing even in complex files.
+
+**Accurate and Optimized Type Detection**
+
+The type for each column is inferred based on the lowest required type from the following ordered list:
+
+`logical` < `integer` < `integer64` < `double` < `character`
+
+This ensures:
+
+- Single up-front allocation of memory using the correct type
+- Avoidance of rereading the file or manually setting colClasses
+- Improved speed and memory efficiency
+
+**Out-of-Sample Type Exceptions**
+
+If a type change occurs outside the sampled rows, `fread()` automatically detects it and rereads only the affected columns from the beginning. This ensures correct type assignment without requiring user intervention. For example, a column sampled as integer might later contain `00A` — triggering an automatic reread as character.
```

Comment:
that sounds nice in theory, but each thread reads some row chunks so it would be really hard to only read the affected columns